sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -96,7 +105,41 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
     programOffering.parentOffering = programOffering;
 
     try {
-      return await this.repo.save(programOffering);
+      const institutionLocation = await this.institutionLocationRepo.findOne({
+        select: {
+          id: true,
+          name: true,
+          institution: {
+            id: true,
+            operatingName: true,
+            legalOperatingName: true,
+            primaryEmail: true,
+          },
+        },
+        relations: { institution: true },
+        where: { id: programOffering.institutionLocation.id },
+      });
+      const institution = institutionLocation.institution;
+      const ministryNotification: InstitutionAddsPendingOfferingNotificationForMinistry =
+        {
+          institutionName: institution.legalOperatingName,
+          institutionOperatingName: institution.operatingName,
+          institutionLocationName: institutionLocation.name,
+          programName: educationProgramOffering.programContext.name,
+          offeringName: educationProgramOffering.offeringName,
+          institutionPrimaryEmail: institution.primaryEmail,
+        };
+      return await this.dataSource.transaction(async (entityManager) => {
+        if (programOffering.offeringStatus === OfferingStatus.CreationPending) {
+          await this.notificationActionsService.saveInstitutionAddsPendingOfferingNotificationForMinistry(
+            ministryNotification,
+            entityManager,
+          );
+        }
```

Comment:
This can be moved to a private method to avoid extending too much the the method that has the main intention of saving the program.